crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -458,23 +454,30 @@ impl<'a> Compressor<'a> {
                 Ty::Str => {
                     let left_string = get_side_free_string_value(left_expr);
                     let right_string = get_side_free_string_value(right_expr);
-                    if let Some(left_string) = left_string && let Some(right_string) = right_string {
+                    if let (Some(left_string), Some(right_string)) = (left_string, right_string) {
                         // In JS, browsers parse \v differently. So do not compare strings if one contains \v.
                         if left_string.contains('\u{000B}') || right_string.contains('\u{000B}') {
                             return Tri::Unknown;
                         }
 
-                        return Tri::for_boolean(left_string == right_string)
+                        return Tri::for_boolean(left_string == right_string);
                     }
 
-                    if let (Expression::UnaryExpression(left), Expression::UnaryExpression(right)) = (left_expr, right_expr)
-                        && (left.operator, right.operator) == (UnaryOperator::Typeof, UnaryOperator::Typeof)
-                        && let Expression::Identifier(left) = &left.argument
-                        && let Expression::Identifier(right) = &right.argument
-                        && left.name == right.name
+                    if let (Expression::UnaryExpression(left), Expression::UnaryExpression(right)) =
+                        (left_expr, right_expr)
                     {
-                        // Special case, typeof a == typeof a is always true.
-                        return Tri::True;
+                        if (left.operator, right.operator)
```

Comment:
This is identical to line 380. Perhaps it should be factored into a function. If it's in a function, early returns might make more sense. Thoughts @Boshen?